### DISCUSSION TITLE
Jasmin Realtime

### DISCUSSION THREAD

**Discussion by BlackOrder** [jookies/jasmin#1098] [Ideas]
Hi, I wrote this python package [jasmin_realtime](https://github.com/BlackOrder/jasmin_realtime) for myself. This is my first Python project. I usually code in PHP :P. I wanted to sync my configurations across multiple instances of Jasmin.
This package will connect a MongoDB cluster to Jasmin. all configurations can live in the MongoDB.
The connection between this package and Jasmin is separated in another package: [jasmin_telnet](https://github.com/BlackOrder/jasmin_telnet) which I wrote some of it and collected most of from: [jasmin-api](https://github.com/jookies/jasmin-api) and [jasmin-web-panel](https://github.com/101t/jasmin-web-panel).

Do you think it's better to connect through telnet or Prospective Proxies. I tried to find documentation for the pb, but couldn't find the `filter` module part. So, I had to write the telnet package.

**Comment by farirat** [comment]
Hello @BlackOrder !
I salute your effort and your sharing-back to the community !
I've been checking both repos and here's my comments:
- It's a great idea to sync cluster instances configurations to a mongo db, if you are going to exclusively manage your jasmin instances through mongo db then you'll need to forbid access to jcli to avoid getting out of sync,
- There still some work to make test coverage, cleanup of the code, documentation, and some deployment examples through k8s and docker,
- Handling the billing stuff is more complex than handling config synchronization, the latter can be done in a one-way synchronization (database > jasmin(s)) where billing is a two-way synchronization (database <---> jasmin(s)) because one user can be charged 5 pdus in instance A then 10 pddus in instance B, in such scenarios, the balance must be centralized and jasmin nodes must inc/dec the balance simultaneously, in such cases, i always deploy interception logics and make third-party/external billing module,
Some other suggestions:
- jasmin-mongo-configuration is a better name than jasmin_realtime
- make the logger feature a separate module, it has nothing to do with the config sync stuff
And yes, telnet is better/easier than pb.
